File: MMDPG.py
import time
import uuid
import pyautogui
import matplotlib.pyplot as plt

# Set the duration of movement detection to 30 minutes
duration = 30 * 60

# Get a unique ID for this session
session_id = uuid.uuid4()

# Initialize lists to store the mouse coordinates
x_values = []
y_values = []

# Get the starting position of the mouse
start_x, start_y = pyautogui.position()

# Get the starting time of the movement detection
start_time = time.time()

# Loop until the duration is reached
while time.time() - start_time < duration:
    # Get the current position of the mouse
    current_x, current_y = pyautogui.position()
    
    # Check if the mouse has moved
    if current_x != start_x or current_y != start_y:
        # Store the new mouse coordinates
        x_values.append(current_x)
        y_values.append(current_y)
        
        # Update the starting position of the mouse
        start_x, start_y = current_x, current_y
        
    # Wait for 0.1 seconds before checking again
    time.sleep(0.1)
    
# Generate a unique filename for the graph
filename = f'movement_pattern_{session_id}.png'

# Plot the mouse movement as a line graph
plt.plot(x_values, y_values)
plt.title('Mouse Movement Pattern')
plt.xlabel('X Coordinate')
plt.ylabel('Y Coordinate')
plt.savefig(filename)

# Print out the filename of the generated graph
print(f'Movement pattern graph saved as {filename}')

File: MMDPG.lua
-- Load the library to interact with the operating system
local os = require("os")

-- Load the library to interact with the mouse
local mouse = require("mouse")

-- Define the time limit in seconds
local time_limit = 1800 -- 30 minutes

-- Define the time interval to check for movement in seconds
local check_interval = 1

-- Define the threshold for movement in pixels
local movement_threshold = 5

-- Generate a unique ID for this run
local id = os.time()

-- Keep track of the mouse position
local last_position = mouse.position()

-- Keep track of the time since last movement
local time_since_movement = 0

-- Create a table to store the movement data
local movement_data = {}

-- Loop until the time limit is reached
while time_since_movement < time_limit do
    -- Wait for the check interval
    os.sleep(check_interval)
    
    -- Get the current mouse position
    local current_position = mouse.position()
    
    -- Calculate the distance between the current and last position
    local distance = math.sqrt((current_position.x - last_position.x)^2 + (current_position.y - last_position.y)^2)
    
    -- If the distance is greater than the threshold, record the movement and reset the time since last movement
    if distance > movement_threshold then
        table.insert(movement_data, distance)
        time_since_movement = 0
    else
        time_since_movement = time_since_movement + check_interval
    end
    
    -- Update the last position
    last_position = current_position
end

-- Print the movement data in a graph
print("Movement data for ID #"..id)
for i, distance in ipairs(movement_data) do
    print(i.."\t"..string.rep("*", distance))
end
